src/extension.ts

```diff
@@ -44,13 +45,13 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
     registerUIExtensionVariables(ext);
     registerAzureUtilsExtensionVariables(ext);
 
-    const refreshAzureTreeEmitter = new vscode.EventEmitter<void>();
+    const refreshAzureTreeEmitter = new vscode.EventEmitter<void | ResourceGroupsItem | ResourceGroupsItem[] | null | undefined>();
```

Comment:
I didn't see any changes that would accommodate all these new data types. Do we have a function to handle all of these different types? I haven't been able to find it anywhere.

Reply:
https://github.com/microsoft/vscode-azureresourcegroups/pull/460/files#diff-5ca32a8fc22d6ae9e10d28b887089aca193084aaf7b5b9dc3d5089b650741fe9L58

Reply:
Yeah, but even if it fires the data, doesn't the event handler need to actually handle all the types? Does the tools package already handle that case?

Reply:
VS Code listens to the `onDidChangeTreeData` event on the tree data provider.